### ISSUE TITLE
Disabled/Read Only Functionality

### ISSUE THREAD

**Issue by lucasdina** [icemanbsi/searchable_dropdown#15]
Is there any way to disable the dropdown? I noticed the disabled hint but no way to disable the dropdown itself. 

Case:
I have a screen with several dropdowns thats reused in multiple places. Sometimes, I need to show whats selected in one dropdown and maintain that value without letting the user change it.

**Comment by lcuis** [comment]
Hello @lucasdina ,

The disabled hint is displayed when `_enabled` is false:
```
  bool get _enabled =>
      widget.items != null &&
      widget.items.isNotEmpty &&
      widget.onChanged != null;
```
Here is an example of disabled Widget based on PR #13 :
```
SearchChoices.single(
        items: [
          DropdownMenuItem(
            child: Text(
                "one item"),
            value:
            "one item",
          )
        ],
        value: "one item",
        hint: "Select one",
        searchHint: "Select one",
        disabledHint: "Disabled",
        onChanged: null,
        dialogBox: true,
        isExpanded: true,
      );
```
Here is the result:
![image](https://user-images.githubusercontent.com/32125299/75432566-e145f180-594e-11ea-90cc-b1c51613dd94.png)

So, the disabled hint is displayed but not the selected value.

This made me realize that the clear button is still enabled when the dropdown is disabled. I just corrected that on the next PR #16 for this plugin. (If you don't want to wait for the publication of this PR, I also implemented it on [my version of the plugin](https://pub.dev/packages/search_choices) ).
Here is the result:
![image](https://user-images.githubusercontent.com/32125299/75461584-4fa4a700-5983-11ea-8401-d04f47ba0472.png)

I also added a `readOnly` bool parameter defaulted to false with the following result if set to true:
![image](https://user-images.githubusercontent.com/32125299/75465201-934ddf80-5988-11ea-9d71-c99a31474c25.png)
